resharper/resharper-unity/src/CSharp/Daemon/UsageChecking/UsageInspectionsSuppressor.cs

```diff
@@ -128,6 +135,20 @@ public bool SuppressUsageInspectionsOnElement(IDeclaredElement element, out Impl
             return false;
         }
 
+        private static bool IsSettingsProvider(IMethod method)
+        {
+            if (method.HasAttributeInstance(KnownTypes.SettingsProviderAttribute, AttributesSource.All) && method.IsStatic)
+            {
+                if (method.ReturnType.IsImplicitlyConvertibleTo(TypeFactory.CreateTypeByCLRName(KnownTypes.SettingsProvider, method.Module),
+                    new XamlWinRTTypeConversionRule(method.Module)))
```

Comment:
XamlWinRTTypeConversionRule is incorrect here, please use `method.Module.GetTypeConversionRule()`

Reply:
You should be able to do something like:

```
(method.ReturnType as IDeclaredType)?.GetTypeElement()?.DerivesFrom(clrTypeName);
```

Although possibly not all on one line :)